## 11 - As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contraram para desenvolver o programa que calculará os reajustes.

## Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:

In [38]:
salario = float(input("Informe o seu salário: "))

Informe o seu salário: 100.10


### salários até RS 280,00 (incluindo) : aumento de 20%

### salários entre RS 280,00 e RS 700,00 : aumento de 15%

### salários entre RS 700,00 e RS 1500,00 : aumento de 10%

### salários de RS 1500,00 em diante : aumento de 5% Após o aumento ser realizado, informe na tela:

### o salário antes do reajuste;

### o percentual de aumento aplicado;

### o valor do aumento;

### o novo salário, após o aumento.

In [39]:
if salario <= 280:
    salario_novo = round((salario * 1.20), 2)
elif salario <= 700:
    salario_novo = round((salario * 1.15), 2)
elif salario <= 1500:
    salario_novo = round((salario * 1.10), 2)
else:
    salario_novo = round((salario * 1.05), 2)
    
print(f"Salário antes do reajuste: R$ {salario}")
print(f"Salário após o reajuste: R$ {salario_novo}")
print(f"Valor do aumento: R$ {round((salario_novo - salario), 2)}")
print(f"Percentual de aumento: {round((((salario_novo - salario)/salario) * 100), 2)} %")

Salário antes do reajuste: R$ 100.1
Salário após o reajuste: R$ 120.12
Valor do aumento: R$ 20.02
Percentual de aumento: 20.0 %


## 12 - Faça um programa para o cálculo de uma folha de pagamento, sabendo que os descontos são do Imposto de Renda, que depende do salário bruto (conforme tabela abaixo) e 3% para o Sindicato e que o FGTS corresponde a 11% do Salário Bruto, mas não é descontado (é a empresa que deposita). O Salário Líquido corresponde ao Salário Bruto menos os descontos. O programa deverá pedir ao usuário o valor da sua hora e a quantidade de horas trabalhadas no mês.

In [1]:
valor_hora = float(input("Informe o valor da sua hora: "))
horas_trabalhadas = float(input("Informe a quantidade de horas trabalhadas: "))
salario_bruto = valor_hora * horas_trabalhadas
print(f"Valor do salário bruto: R$ {salario_bruto}")

Informe o valor da sua hora: 5
Informe a quantidade de horas trabalhadas: 220
Valor do salário bruto: R$ 1100.0


### Desconto do IR:

### Salário Bruto até 900 (inclusive) - isento

### Salário Bruto até 1500 (inclusive) - desconto de 5%

### Salário Bruto até 2500 (inclusive) - desconto de 10%

### Salário Bruto acima de 2500 - desconto de 20% Imprima na tela as informações, dispostas conforme o exemplo abaixo. No exemplo o valor da hora é 5 e a quantidade de hora é 220.

###        Salário Bruto: (5 * 220)        : RS 1100,00
###        (-) IR (5%)                     : RS   55,00  
###        (-) INSS ( 10%)                 : RS  110,00
###        FGTS (11%)                      : RS  121,00
###        Total de descontos              : RS  165,00
###        Salário Liquido                 : RS  935,00

In [10]:
if salario_bruto > 900 and salario_bruto <= 1500:
    ir = 5
elif salario_bruto <= 2500:
    ir = 10
else:
    ir = 20
    
print(f"Desconto de IR ({ir}%): R$ {((salario_bruto * ir)/100)}")
print(f"Desconto de INSS (10%): R$ {salario_bruto * 0.1}")
print(f"FGTS (11%): R$ {salario_bruto * 0.11}")
print(f"Total de descontos: {((salario_bruto * ir)/100) + (salario_bruto * 0.1)}")
salario_liquido = salario_bruto - (((salario_bruto * ir)/100) + (salario_bruto * 0.1))
print(f"Salário líquido: R$ {salario_liquido}")

Desconto de IR (5%): R$ 55.0
Desconto de INSS (10%): R$ 110.0
FGTS (11%): R$ 121.0
Total de descontos: 165.0
Salário líquido: R$ 935.0


## 13 - Faça um Programa que leia um número e exiba o dia correspondente da semana. (1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [16]:
dia = input("Digite um número de 1 a 7: ")

def dias_semana(d):
    switcher = {
        '1': 'Domingo',
        '2': 'Segunda-feira',
        '3': 'Terça-feira',
        '4': 'Quarta-feira',
        '5': 'Quinta-feira',
        '6': 'Sexta-feira',
        '7': 'Sábado'
    }
    return switcher.get(d, "Valor inválido")

dias_semana(dia)

Digite um número de 1 a 7: a


'Valor inválido'

## 14 - Faça um programa que lê as duas notas parciais obtidas por um aluno numa disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos obedece à tabela abaixo:

###  Média de Aproveitamento  Conceito
###  Entre 9.0 e 10.0        A
###  Entre 7.5 e 9.0         B
###  Entre 6.0 e 7.5         C
###  Entre 4.0 e 6.0         D
###  Entre 4.0 e zero        E

### O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito for D ou E.

In [10]:
n1 = float(input("Digite a primeira nota: "))
n2 = float(input("Digite a segunda nota: "))
media = (n1 + n2)/2

conceito = "A"

if media <= 4:
    conceito = "E"
elif media <= 6:
    conceito = "D"
elif media <= 7.5:
    conceito = "C"
elif media <= 9:
    conceito = "B"
else:
    conceito = "A"
    
def mensagem(c):
    switcher = {
        "A":"APROVADO",
        "B":"APROVADO",
        "C":"REPROVADO",
        "D":"REPROVADO",
        "E":"REPROVADO"
    }
    return switcher.get(c, "VALOR INVÁLIDO!")

situacao = mensagem(conceito)

print(f"\nMédia: {media}\nConceito: {conceito}\nSituação: {situacao}")

Digite a primeira nota: 6
Digite a segunda nota: 7

Média: 6.5
Conceito: C
Situação: REPROVADO


## 15 - Faça um Programa que peça os 3 lados de um triângulo. O programa deverá informar se os valores podem ser um triângulo. Indique, caso os lados formem um triângulo, se o mesmo é: equilátero, isósceles ou escaleno.

### Dicas:

### Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;

### Triângulo Equilátero: três lados iguais;

### Triângulo Isósceles: quaisquer dois lados iguais;

### Triângulo Escaleno: três lados diferentes;

In [44]:
try:
    a = float(input("Digite o primeiro lado do triângulo: "))
    b = float(input("Digite o segundo lado do triângulo: "))
    c = float(input("Digite o terceiro lado do triângulo: "))
except ValueError:
    a,b,c=0,0,0
    print("Digite apenas números!")

tipo=""
    
if a<=0 or b<=0 or c<=0:
    print("Não são permitidos valores nulos, negativos ou caracteres.")
elif a+b>c and a+c>b and b+c>a:
    if a==b==c:
        tipo="Equilátero"
    elif a==b or a==c or b==c:
        tipo="Isósceles"
    else:
        tipo="Escaleno"
else:
    print("Não é um triângulo!")

print(tipo)

Digite o primeiro lado do triângulo: s
Digite apenas números!
Não são permitidos valores nulos, negativos ou caracteres.



## 16 - Faça um programa que calcule as raízes de uma equação do segundo grau, na forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as consistências, informando ao usuário nas seguintes situações:

In [30]:
#print("Delta > 0, a=1, b=-3, c=-10\nDelta = 0, a=9, b=-12, c=4")
try:
    a = float(input("Digite o valor de a: "))
    if a==0:
        print("A equação não é do segundo grau, pois a é igual a 0.")
    else:
        b = float(input("Digite o valor de b: "))
        c = float(input("Digite o valor de c: "))
except ValueError:
    print("Digite apenas números!")

Digite o valor de a: 1
Digite o valor de b: -3
Digite o valor de c: -10


### a) Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e o programa não deve fazer pedir os demais valores, sendo encerrado;

### b) Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao usuário e encerre o programa;

### c) Se o delta calculado for igual a zero a equação possui apenas uma raiz real; informe-a ao usuário;

### d) Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [31]:
delta = (b**2) - (4*(a*c))
print(f"O valor de delta é: {delta}")
    
if delta < 0:
    print("A equação não possui raízes reais, pois o delta é negativo.")
elif delta == 0:
    x = float(-b/(2*a))
    print(f"x = {x}")
else:
    x1 = float((-b + (delta**(1/2)))/(2*a))
    x2 = float((-b - (delta**(1/2)))/(2*a))
    print(f"x'  = {x1}\nx'' = {x2}")

O valor de delta é: 49.0
x'  = 5.0
x'' = -2.0


## 17 - Faça um Programa que peça um número correspondente a um determinado ano e em seguida informe se este ano é ou não bissexto.

In [45]:
try:
    ano = int(input("Digite um ano: "))
except ValueError:
    print("Digite apenas números!")
    
if ano <= 0:
    print("Não são aceitos números nulos ou negativos!")
elif (ano > 0) and ((ano % 4 == 0) and (ano % 100 != 0)) or (ano % 400 == 0):
    print(f"{ano} é bissexto.")
else:
    print(f"{ano} não é bissexto.")

Digite um ano: 100
100 não é bissexto.


## 18 - Faça um Programa que peça uma data no formato dd/mm/aaaa e determine se a mesma é uma data válida.

In [19]:
from datetime import date

data = date.today()
format_data = '{}/{}/{}'.format(data.day, data.month, data.year)

print(f"Exemplo:\n{format_data}")

print("___________________________\n")

try:
    d = date(input("Digite uma data (dd/mm/aaaa): "))
except TypeError:
    print("Digite a data no formato correto!")


Exemplo:
31/12/2019
___________________________

Digite uma data (dd/mm/aaaa): 121


## 19 - Faça um Programa que leia um número inteiro menor que 1000 e imprima a quantidade de centenas, dezenas e unidades do mesmo.

### Observando os termos no plural a colocação do "e", da vírgula entre outros. Exemplo:

### 326 = 3 centenas, 2 dezenas e 6 unidades

### 12 = 1 dezena e 2 unidades Testar com: 326, 300, 100, 320, 310,305, 301, 101, 311, 111, 25, 20, 10, 21, 11, 1, 7 e 16

## 20 - Faça um Programa para leitura de três notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e presentar:

### a) A mensagem "Aprovado", se a média for maior ou igual a 7, com a respectiva média alcançada;

### b) A mensagem "Reprovado", se a média for menor do que 7, com a respectiva média alcançada;

### c) A mensagem "Aprovado com Distinção", se a média for igual a 10.